src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,298 @@
+                    tasks.add(t);
+                    if (splitLine[1].equals("1")) {
+                        t.markAsDone();
+                    }
+                }
+            }
+            fileReader.close();
+        } catch (FileNotFoundException e) {
+            ;
+        } catch (IOException e) {
+            System.out.println("\tAn error occured while trying to create a data directory :(");
+            //System.out.println(e.getMessage());
+        }
+    }
+
+    public static boolean saveData() {
+        try {
+            System.out.println("\tSaving your data...");
+            FileWriter fileWriter = new FileWriter("data/duke.txt");
+            for (int i=0; i<tasks.size(); i++) {
+                Task currentTask = tasks.get(i);
+                fileWriter.write(currentTask.getType() + " | " + ((currentTask.isDone()) ? "1" : "0") + " | "
+                        + currentTask.getDescription() + ((currentTask.hasDate()) ? " | " + currentTask.getDate() : "") + '\n' );
+            }
+            fileWriter.close();
+            System.out.println("\tData saved");
+            return true;
+        } catch (IOException e) {
+            System.out.println("\tAn error occurred while saving your data :(");
+            return false;
+        }
+    }
+
+    public static void takeInput() {
+        Scanner userInput = new Scanner(System.in);
+        while (userInput.hasNextLine()) {
+            String command = userInput.nextLine();
+            if (command.equals("bye") || command.equals("exit")) {
+                dividerLine();
+                if (exitDuke()) {
+                    dividerLine();
+                    userInput.close();
+                    break;
+                }
+                dividerLine();
+            } else if (command.equals("list")) {
+                dividerLine();
+                list();
+                dividerLine();
+            } else if (command.startsWith("done")) {
+                dividerLine();
+                finishTask(command);
+                dividerLine();
+            } else if (command.startsWith("todo")) {
+                dividerLine();
+                addTodo(command);
+                dividerLine();
+            } else if (command.startsWith("event")) {
+                dividerLine();
+                addEvent(command);
+                dividerLine();
+            } else if (command.startsWith("deadline")) {
+                dividerLine();
+                addDeadline(command);
+                dividerLine();
+            } else if (command.startsWith("delete")) {
+                dividerLine();
+                deleteTask(command);
+                dividerLine();
+            } else if (command.equals("save")) {
+                dividerLine();
+                saveData();
+                dividerLine();
+            } else {
+                dividerLine();
+                unknownCommand();
+                dividerLine();
+            }
```

Comment:
Consider extracting out dividerLine() to the beginning and the end of the while loop. This will reduce redundancy and the number of lines in this function.